From 6e1c19ae881dc117e1c47ec6f9e03e46175d95bc Mon Sep 17 00:00:00 2001
From: Michael Donahoe <michael.donahoe@systematicgroup.com>
Date: Fri, 12 Jul 2019 17:22:27 +0000
Subject: [PATCH 2/4] update onKeyStatusesChange

---
 MediaSystem.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MediaSystem.cpp b/MediaSystem.cpp
index 6124309..de350d0 100644
--- a/MediaSystem.cpp
+++ b/MediaSystem.cpp
@@ -169,7 +169,7 @@ public:
         _adminLock.Unlock();
     }
 
-    virtual void onKeyStatusesChange(const std::string& session_id) {
+    virtual void onKeyStatusesChange(const std::string& session_id, bool has_new_usable_key) {
 
         _adminLock.Lock();
 
-- 
2.7.4

