From 14a50e6a4676818be1448e1f19ea933fc80fa1e8 Mon Sep 17 00:00:00 2001
From: Michael Donahoe <michael.donahoe@systematicgroup.com>
Date: Wed, 17 Apr 2019 15:02:15 +0000
Subject: [PATCH] NXP patch for deleting previous decrypt buffer

---
 MediaSession.cpp | 13 ++++++++++---
 MediaSession.h   |  2 ++
 2 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/MediaSession.cpp b/MediaSession.cpp
index 12849df..65f4c76 100644
--- a/MediaSession.cpp
+++ b/MediaSession.cpp
@@ -204,10 +204,10 @@ void MediaKeySession::Update(
       f_cbKeyMessageResponse);
   if (widevine::Cdm::kSuccess != m_cdm->update(m_sessionId, keyResponse))
   { 
-     fprintf(stderr,"cdm->update success\n");
+     fprintf(stderr,"cdm->update error\n");
      onKeyStatusChange();
   } else {
-     fprintf(stderr,"cdm->update error\n");
+     fprintf(stderr,"cdm->update success\n");
   }
 }
 
@@ -319,12 +319,19 @@ CDMi_RESULT MediaKeySession::Decrypt(
       input.key_id_length = (it->first).size();
       input.iv = m_IV;
       input.iv_length = sizeof(m_IV);
+      
+      if(m_DecyrptOut != NULL)
+      {
+        free(m_DecyrptOut);  //Free previous buffer
+      }
+
 
       if (widevine::Cdm::kSuccess == m_cdm->decrypt(input, output)) {
         /* Return clear content */
         *f_pcbOpaqueClearContent = output.data_length;
         *f_ppbOpaqueClearContent = outputBuffer;
-        
+  	
+	m_DecyrptOut = outputBuffer; 
         status = CDMi_SUCCESS;
       }
     }
diff --git a/MediaSession.h b/MediaSession.h
index d64d4c2..b49a6fc 100644
--- a/MediaSession.h
+++ b/MediaSession.h
@@ -98,6 +98,8 @@ private:
     IMediaKeySessionCallback *m_piCallback;
     uint8_t m_IV[16];
     CurlPost m_curl;
+    uint8_t *m_DecyrptOut;
+    uint8_t m_tempValue[16];
 };
 
 }  // namespace CDMi
-- 
2.7.4

